lib/async_hooks.js

```diff
@@ -255,29 +255,24 @@ class AsyncLocalStorage {
     resource[this.kResourceStore] = store;
   }
 
-  _exit() {
-    const resource = executionAsyncResource();
-    if (resource) {
-      resource[this.kResourceStore] = undefined;
-    }
-  }
-
-  runSyncAndReturn(store, callback, ...args) {
-    this._enter(store);
-    try {
+  run(store, callback, ...args) {
+    const resource = new AsyncResource('AsyncLocalStorage');
```

Comment:
Can we benchmark the impact of having multiple `AsyncLocalStorages` calling `run(...)`?
One of the advantages of `runSyncAndReturn(...)` is the predictible performance. Here we lose this.